compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -170,19 +164,12 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
         bc.aload(elementType)
       }
       else if (isArraySet(code)) {
-        args match {
-          case a1 :: a2 :: Nil =>
-            genLoad(a1, INT)
-            genLoad(a2)
-            // the following line should really be here, but because of bugs in erasure
-            // we pretend we generate whatever type is expected from us.
-            //generatedType = UNIT
-            bc.astore(elementType)
-          case _ =>
-            abort(s"Too many arguments for array set operation: $tree")
-        }
-      }
-      else {
+        val List(a1, a2) = args
```

Comment:
What happens if there is a different number of arguments? I think it is best to handle this case as was done previously with the `abort` call so that to avoid confusing error messages.

Reply:
> What happens if there is a different number of arguments?
 
If it happens there's a bug somewhere else in the backend.

> I think it is best to handle this case as was done previously with the abort call

I'm not opposed to this change in theory, but I don't want to introduce unnecessary changes compared to the scalac backend: every such change makes it harder to port code between the two backends. So I'd like to keep this as is unless this is also changed in scalac.